### PR TITLE
Migrate to commercetools conventions

### LINKED ISSUE #44: Migrate to commercetools Node.js library conventions

### PR DESCRIPTION
**Pull request by hisabimbola**
#### Summary

Migrate sphere-product-type-export to commercetools NodeJS conventions

Fixes #44 
#### Reviewers

- [ ] Reviewer 1 has approved the PR
- [ ] Reviewer 2 has approved the PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,28 +7,30 @@
...
+    "clean": "rimraf dist coverage",
+    "commit": "git-cz",
+    "lint": "./scripts/lint lib test",
+    "test": "npm run lint && cross-env NODE_ENV=test babel-node node_modules/.bin/tape test/**/*.spec.js test/*.spec.js | tap-spec",
+    "check-coverage": "istanbul check-coverage --statements 100 --branches 100 --functions 100 --lines 100",
+    "report-coverage": "cat ./coverage/lcov.info | codecov",
     "test:watch": "npm test -- --watch",
-    "test:features": "npm run build && babel-node $(npm bin)/cucumber.js"
+    "test:features": "npm run build && babel-node $(npm bin)/cucumber.js",
+    "test:ci": "npm run clean && cross-env NODE_ENV=test babel-node node_modules/.bin/babel-istanbul cover node_modules/.bin/tape -- test/**/*.js"
```
Comment (Siilwyn) [inline package.json]: Why is there a separate CI test command?

File: package.json [hunk 2]
```diff
@@ -37,15 +39,22 @@
...
   "engines": {
     "node": ">= 0.10"
   },
+  "config": {
+    "commitizen": {
+      "path": "node_modules/cz-conventional-changelog"
+    },
+    "ghooks": {
+      "commit-msg": "validate-commit-msg",
+      "pre-commit": "./scripts/lint-staged-files && npm test"
```
Comment (Siilwyn) [inline package.json]: I wouldn't like to wait for the tests to finish before each commit, thoughts?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+<!-- Provide a short summary of your changes -->
+
+#### Description
+<!-- Describe the changes in this PR here and provide some context -->
+
+#### Reviewers
+<!-- The persons assigned to review this PR should mark their item as done when they are happy with this PR to be merged -->
+
+- [ ] Reviewer 1 has approved the PR
+- [ ] Reviewer 2 has approved the PR
```
Comment (emmenko) [inline .github/PULL_REQUEST_TEMPLATE.md]: I guess we can remove this section. See https://github.com/sphereio/sphere-product-type-export/pull/4/files#r82736614

File: .travis.yml [hunk 4]
```diff
@@ -1,15 +1,17 @@
 language: node_js
 node_js:
-  - '0.12'
   - '4.4.0'
```
Comment (emmenko) [inline .travis.yml]: PS: I guess we can simply use `4` here

File: package.json (hunk 2 above)
Comment (emmenko) [inline package.json]: I would also prefer to only lint the files.

File: test/commands/import.spec.js [hunk 5]
```diff
@@ -1,214 +1,236 @@
...
+    const spy2 = sinon.stub(command, '_preProcess', opts => command._process(
+      Object.assign(opts, { config: {}, credentials: fakeCredentials }))
+    )
     command.run(['node', `${BIN_DIR}/sphere-import`,
       '-p', 'foo', '-t', 'stock', '-f', './foo.json'])
-    command.program.project.should.be.equal('foo')
-    command.program.type.should.be.equal('stock')
-    command.program.from.should.be.equal('./foo.json')
-    expect(spy1.args[0][0]).toEqual({
+    t.equal(command.program.project, 'foo')
```
Comment (emmenko) [inline test/commands/import.spec.js]: In general it's good to provide a message as third param explaining what the assertion does

File: test/utils/command.spec.js [hunk 6]
```diff
@@ -1,48 +1,60 @@
...
-  it('should throw when calling run', () => {
-    expect(() => command.run())
-      .toThrow('Base run method must be overridden')
+test(`Command
+  should throw when calling run`, (t) => {
+  before().then((command) => {
+    t.throws((() => command.run()), 'Base run method must be overridden')
+    t.end()
   })
+})
```
Comment (emmenko) [inline test/utils/command.spec.js]: You're missing the catch handler here

**Pull request merged**